torch/distributed/checkpoint/filesystem.py

```diff
@@ -434,6 +435,16 @@ def __init__(
         self.thread_count = thread_count
         self.per_thread_copy_ahead = per_thread_copy_ahead
 
+    def stage(self, state_dict: STATE_DICT_TYPE) -> STATE_DICT_TYPE:
```

Comment:
Right now the planner component support resolve_tensor and resolve_bytes APIs to lookup write plan data.

When consuming staged data, the storage_writer should be able to consume ANY type of data (e.g., file path, shm-handle). Therefore, we probably need to update the resolve_data in planner;

def resolve_data(write_item) -> Any:

thoughts?

Reply:
I think this might be worth another issue or PR. Let me know if that's something you're interested in :)